### PR TITLE
Added "Fixes" section for 1.14 release notes

### PR DESCRIPTION
**Pull request by tvansteenburgh**
Definitely needs a review. This is a curated list but there may yet be cruft, dupes, and typos. We might even want to add some of the "Added" items up to the What's new section if they seem interesting enough.

@Cynerva and @evilnick please have a look.

### CODE REVIEW

File: pages/k8s/release-notes.md [hunk 1]
```diff
@@ -45,16 +45,63 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
-xxxxxx xxx xxx xx xxxx xxxx  [documentation][doc-link].
+- one-liner xxxxxx xxxxx xxx
+- one-liner xxx x xx x xx xxx
 
 ## Fixes
 
-- Fixed xxxx xxx xx xx xxx xxxxxxx xxxx ([Issue](https://bugs.launchpad.net/charmed-kubernetes/))
-- Fixed xxxx xxx xxxxx xx xx xxxxx xxxx ([Issue](https://bugs.launchpad.net/charmed-kubernetes/))
-- Fixed xx x xxx x xx xxx xxx xxxx xx xx ([Issue](https://bugs.launchpad.net/charmed-kubernetes/))
+ - Added an action to upgrade Docker ([Issue](https://github.com/juju-solutions/layer-docker/pull/135))
```
Comment (evilnick) [inline pages/k8s/release-notes.md]: would this be worth adding to the main notes?

File: pages/k8s/release-notes.md [hunk 2]
```diff
@@ -45,16 +45,63 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Added better multi-client support to EasyRSA ([Issue](https://github.com/juju-solutions/layer-easyrsa/pull/15))
+ - Added block storage config options ([Issue](https://github.com/juju-solutions/charm-openstack-integrator/pull/15))
+ - Added block storage options for OpenStack ([Issue](https://github.com/juju-solutions/kubernetes/pull/218))
+ - Added dashboard-auth config option to master ([Issue](https://github.com/juju-solutions/kubernetes/pull/222))
+ - Added dashboard login using kubeconfig ([Issue](https://github.com/juju-solutions/cdk-addons/pull/72))
+ - Added docker registry handling to master ([Issue](https://github.com/juju-solutions/kubernetes/pull/210))
+ - Added more TLS options to Docker registry ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/20))
+ - Added new folder/respool_path config for vSphere ([Issue](https://github.com/juju-solutions/charm-vsphere-integrator/pull/2))
+ - Added proxy support to Docker registry ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/21))
```
Comment (evilnick) [inline pages/k8s/release-notes.md]: ^ this also seems like it would be important to some people

File: pages/k8s/release-notes.md [hunk 3]
```diff
@@ -45,16 +45,63 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Added read-only mode for Docker registry ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/22))
+ - Added token auth option for dashboard ([Issue](https://github.com/juju-solutions/cdk-addons/pull/77))
```
Comment (evilnick) [inline pages/k8s/release-notes.md]: ^ ditto

File: pages/k8s/release-notes.md (hunk 1 above)
Comment (tvansteenburgh) [reply pages/k8s/release-notes.md]: Yes, please see the release announcement doc where I included this in the Features section, and borrow some verbiage from there. Thanks!

File: pages/k8s/release-notes.md (hunk 2 above)
Comment (tvansteenburgh) [reply pages/k8s/release-notes.md]: Ditto the above.

File: pages/k8s/release-notes.md (hunk 3 above)
Comment (tvansteenburgh) [reply pages/k8s/release-notes.md]: This one I think we should leave out of "Features". It's too minor in my opinion.

File: pages/k8s/release-notes.md [hunk 4]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
 
 ## Fixes
 
-- Fixed xxxx xxx xx xx xxx xxxxxxx xxxx ([Issue](https://bugs.launchpad.net/charmed-kubernetes/))
-- Fixed xxxx xxx xxxxx xx xx xxxxx xxxx ([Issue](https://bugs.launchpad.net/charmed-kubernetes/))
-- Fixed xx x xxx x xx xxx xxx xxxx xx xx ([Issue](https://bugs.launchpad.net/charmed-kubernetes/))
+ - Added an action to upgrade Docker ([Issue](https://github.com/juju-solutions/layer-docker/pull/135))
+ - Added better multi-client support to EasyRSA ([Issue](https://github.com/juju-solutions/layer-easyrsa/pull/15))
+ - Added block storage config options ([Issue](https://github.com/juju-solutions/charm-openstack-integrator/pull/15))
+ - Added block storage options for OpenStack ([Issue](https://github.com/juju-solutions/kubernetes/pull/218))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: These two lines are redundant. Different PRs, same issue.

File: pages/k8s/release-notes.md [hunk 5]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Added dashboard-auth config option to master ([Issue](https://github.com/juju-solutions/kubernetes/pull/222))
+ - Added dashboard login using kubeconfig ([Issue](https://github.com/juju-solutions/cdk-addons/pull/72))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: This was covered in a [prior release note](https://github.com/juju-solutions/kubernetes-docs/blob/27adcc19039f2a8f1ebe3bf7c427ea8dfa84fe18/pages/k8s/release-notes-historic.md#cve-2018-18264). I think we can drop it.

File: pages/k8s/release-notes.md (hunk 4 above)
Comment (evilnick) [reply pages/k8s/release-notes.md]: we should just have two issue links in that case (some people track by issue)

File: pages/k8s/release-notes.md [hunk 6]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Added docker registry handling to master ([Issue](https://github.com/juju-solutions/kubernetes/pull/210))
+ - Added more TLS options to Docker registry ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/20))
+ - Added new folder/respool_path config for vSphere ([Issue](https://github.com/juju-solutions/charm-vsphere-integrator/pull/2))
+ - Added proxy support to Docker registry ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/21))
+ - Added read-only mode for Docker registry ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/22))
+ - Added token auth option for dashboard ([Issue](https://github.com/juju-solutions/cdk-addons/pull/77))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: Duplicate of "Added dashboard-auth config option to master" above.

File: pages/k8s/release-notes.md [hunk 7]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Fixed `allow-privileged` not enabled when Ceph relation joins ([Issue](https://github.com/juju-solutions/kubernetes/pull/197))
+ - Fixed apt install source for VaultLocker ([Issue](https://github.com/juju-solutions/layer-vaultlocker/pull/3))
+ - Fixed CDK broken when using GPUs due to nvidia-docker2 dependency failure ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/720))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: Already covered in a [prior release note](https://github.com/juju-solutions/kubernetes-docs/blob/27adcc19039f2a8f1ebe3bf7c427ea8dfa84fe18/pages/k8s/release-notes-historic.md#113-bugfix-release).

File: pages/k8s/release-notes.md [hunk 8]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Fixed Ceph relation join not creating necessary pools ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/631))
+ - Fixed Ceph volume provisioning fails with "No such file or directory" ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/710))
+ - Fixed client certs not having the requested common name ([Issue](https://github.com/juju-solutions/layer-easyrsa/pull/19))
+ - Fixed create_client_cert never running ([Issue](https://github.com/juju-solutions/layer-easyrsa/pull/17))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: These last two fixed issues that were only in edge, never in stable. Do we want to include notes for edge-only issues?

File: pages/k8s/release-notes.md [hunk 9]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Fixed csi-rbdplugin-attacher crash loop when allow-privileged=auto or false ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/674))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: Duplicate of "Fixed `allow-privileged` not enabled when Ceph relation joins"

File: pages/k8s/release-notes.md [hunk 10]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Fixed dashboard CVE-2018-18264 ([Issue](https://github.com/juju-solutions/cdk-addons/pull/68))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: Already covered in a [prior release note](https://github.com/juju-solutions/kubernetes-docs/blob/27adcc19039f2a8f1ebe3bf7c427ea8dfa84fe18/pages/k8s/release-notes-historic.md#cve-2018-18264). Also this is a duplicate of "Added dashboard login using kubeconfig" above.

File: pages/k8s/release-notes.md [hunk 11]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Fixed detecting of changed AppKV values ([Issue](https://github.com/juju-solutions/layer-vault-kv/pull/6))
+ - Fixed docker-ce-version config not working for non-NVIDIA configuration ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/621))
+ - Fixed Docker registry behavior with multiple frontends ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/26))
+ - Fixed Docker registry not cleaning up old relation data ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/28))
+ - Fixed Docker registry to correctly handle frontend removal ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/29))
+ - Fixed Docker registry to work behind a TLS-terminating frontend ([Issue](https://github.com/CanonicalLtd/docker-registry-charm/pull/25))
+ - Fixed error: snap "etcd" is not compatible with --classic ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/727))
+ - Fixed etcd/0 Errored with 0 known peers ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/702))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: Already covered in a [prior release note](https://github.com/juju-solutions/kubernetes-docs/blob/27adcc19039f2a8f1ebe3bf7c427ea8dfa84fe18/pages/k8s/release-notes-historic.md#fixes-1). "Added always restart for etcd"

File: pages/k8s/release-notes.md [hunk 12]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Fixed Keystone not working with multiple masters ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/681))
+ - Fixed kubeconfig should contain the VIP when keepalived used with kubeapi-load-balancer ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/712))
+ - Fixed metrics server for k8s 1.11 ([Issue](https://github.com/juju-solutions/cdk-addons/pull/81))
+ - Fixed proxy var to apply when adding an apt-key ([Issue](https://github.com/juju-solutions/layer-docker/pull/133))
+ - Fixed RBAC enabled results in error: unable to upgrade connection ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/696))
+ - Fixed registry action creating configmap in the wrong namespace  ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/721))
+ - Fixed rules for metrics-server ([Issue](https://github.com/juju-solutions/cdk-addons/pull/76))
+ - Fixed snapd_refresh charm option conflict ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/657))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: Already covered in a [prior release note](https://github.com/juju-solutions/kubernetes-docs/blob/27adcc19039f2a8f1ebe3bf7c427ea8dfa84fe18/pages/k8s/release-notes-historic.md#113-bugfix-release).

File: pages/k8s/release-notes.md [hunk 13]
```diff
@@ -45,16 +45,69 @@ Existing deployments that are upgraded to **CDK 1.14** will
...
+ - Fixed status when writing kubeconfig file ([Issue](https://github.com/juju-solutions/kubernetes/pull/202))
+ - Fixed "subnet not found" to be non-fatal ([Issue](https://github.com/juju-solutions/charm-aws-integrator/pull/34))
+ - Fixed vSphere integrator charm not updating cloud-config when setting new charm defaults  ([Issue](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/695))
+ - Fixed wrong name in requestheader-allowed-names ([Issue](https://github.com/juju-solutions/kubernetes/pull/217))
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: This is another edge-only issue.

**Comment by Cynerva** [comment]
All the links say "Issue" but half of them are pull requests.

File: pages/k8s/release-notes.md (hunk 4 above)
Comment (tvansteenburgh) [reply pages/k8s/release-notes.md]: I wouldn't bother. For the dupes I found, I just deleted one. Github keeps them linked so ppl can connect the dots if they need to, but until this listed is generated completely automatically, we don't really have time to be agonizing over it.

File: pages/k8s/release-notes.md (hunk 8 above)
Comment (tvansteenburgh) [reply pages/k8s/release-notes.md]: Nah, they can be dropped, good catch.

File: pages/k8s/release-notes.md (hunk 13 above)
Comment (tvansteenburgh) [reply pages/k8s/release-notes.md]: Good catch, can remove.

**Comment by tvansteenburgh** [comment]
> All the links say "Issue" but half of them are pull requests.

Yeah, I'm okay with this for now. It was the same for the last release. The script just looks at prs right now, because that yields a more reliable list, but I'm sure we can improve it.

File: pages/k8s/release-notes.md (hunk 4 above)
Comment (evilnick) [reply pages/k8s/release-notes.md]: ok

**Comment by evilnick** [comment]
Okay, as per my previous comment, I am going to remove the release date and bundle info, then merge this, make a new 'current' branch containing the 1.14 data and use that to prepare the mega PR for ubuntu.com. There is still a chance it can be live Monday.

@tvansteenburgh 
If that isn't what you want, text the word 'STOP' via any recognised means of communication I care about.

**Pull request merged**